cpp/src/traversal/bfs_kernels.cuh

```diff
@@ -834,8 +841,8 @@ __global__ void topdown_expand_kernel(
             vec_u[iv]                                    = frontier[k];  // origin of this edge
             vec_frontier_degrees_exclusive_sum_index[iv] = frontier_degrees_exclusive_sum[k];
           } else {
-            vec_u[iv]                                    = -1;
-            vec_frontier_degrees_exclusive_sum_index[iv] = -1;
+            vec_u[iv]                                    = ~IndexType(0);
+            vec_frontier_degrees_exclusive_sum_index[iv] = ~IndexType(0);
```

Comment:
invalid_vertex_id<IndexType>::value is much more readable than ~IndexType(0)

Reply:
Done.